src/wallets/hardware/LedgerWallet.ts

```diff
@@ -72,6 +65,32 @@ export default class LedgerWallet
     this.derivationPath = path;
   }
 
+  async setAccountId(accountId: string) {
+    this.accountId = accountId;
+  }
+
+  async checkAccountId(accountId: string, publicKey: string) {
+    const state = getState();
+    const provider = new providers.JsonRpcProvider(`https://rpc.${state.options.networkId}.near.org`);
+
+    const response: any = await provider
+      .query({
+        request_type: "view_access_key",
+        finality: "final",
+        account_id: accountId,
+        public_key: publicKey,
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+
+    if (response) {
+      this.nonce = response.nonce;
+    }
```

Comment:
This should assign `this.nonce` in the `.then` to avoid potential issues if we returned something other than the implicit `undefined` in the `.catch`:

```
await provider
  .query({
    request_type: "view_access_key",
    finality: "final",
    account_id: accountId,
    public_key: publicKey,
  })
  // We really need to move away from using any, but I'll let this slide
  // for now as it's clearly a project-wide problem right now.
  .then((res: any) => {
    this.nonce = res.nonce;
  })
  .catch((err) => {
    console.log(err);
  });
```

Reply:
I pushed a commit with these changes.